### ISSUE TITLE
Unsuccessful uploads may pollute local blockstore with orphaned blocks

### ISSUE THREAD

**Issue by benbierens** [status-im/nim-codex#368]
codes/node.nim:186
```
  try:
    while (
      let chunk = await chunker.getBytes();
      chunk.len > 0):

      trace "Got data from stream", len = chunk.len
      without blk =? bt.Block.new(chunk):
        return failure("Unable to init block from chunk!")

      blockManifest.add(blk.cid)
      if err =? (await self.blockStore.putBlock(blk)).errorOption:
        trace "Unable to store block", cid = blk.cid, err = err.msg
        return failure(&"Unable to store block {blk.cid}")
```

Should 'self.blockStore.putBlock' fail after several blocks have already been successfully put, the previous blocks will not be removed. Errors may also occur further down this function while the manifest is completed and stored that could cause the same situation.
Are we counting on the blockstore maintenance to eventually clean up these orphaned blocks? Or should be modify this to clean up the blocks in case of failure? (Or am I not seeing something?)

**Comment by dryajov** [comment]
> Are we counting on the blockstore maintenance to eventually clean up these orphaned blocks? Or should be modify this to clean up the blocks in case of failure? (Or am I not seeing something?)

Yes, in general we're counting on the maintenance to clean it up, we should probably stop the upload and cleanup the failed blocks.
